pkg/quota/v1/resources.go

```diff
@@ -199,15 +206,32 @@ func ContainsPrefix(prefixSet []string, item corev1.ResourceName) bool {
 	return false
 }
 
-// Intersection returns the intersection of both list of resources
+// Intersection returns the intersection of both list of resources, deduped and sorted
 func Intersection(a []corev1.ResourceName, b []corev1.ResourceName) []corev1.ResourceName {
-	setA := ToSet(a)
-	setB := ToSet(b)
-	setC := setA.Intersection(setB)
-	result := []corev1.ResourceName{}
-	for _, resourceName := range setC.List() {
-		result = append(result, corev1.ResourceName(resourceName))
+	result := make([]corev1.ResourceName, 0, len(a))
```

Comment:
Current code has complexity O(M*N)

Can we do this:
```
setA := ToSet(a)
for _, item := range b {
  if (setA.Has(item)) {
    result = append(result, item)
  }
}
```
The complexity is O(M+N).

Reply:
I'm not opposed but would like to know the size at which the different becomes meaningful. Avoiding allocations is useful and iterating over small lists is not problematic.

Reply:
This is my understanding:
without dynamic code (which switches algorithm according to input sizes), we should be prepared for large input sizes where performance becomes first concern (the additional memory is linear in input size).

If the input sizes are small, either approach is acceptable.

So I would opt for the version I proposed above.

Reply:
We can change the return type of ResourceNames to map[corev1.ResourceName]struct{}.
This way, by the time Intersection is called, it has two Sets available - facilitating the above algorithms.
@liggitt 
What do you think ?